recipe_scrapers/coopse.py

```diff
@@ -0,0 +1,110 @@
+# mypy: allow-untyped-defs
+import json
+import re
+
+import requests
+
+from recipe_scrapers._grouping_utils import IngredientGroup
+
+from ._abstract import AbstractScraper
+
+SCRIPT_PATTERN = re.compile('"recipeId":"(?P<recipeId>[0-9]+)"')
+
+
+class CoopSE(AbstractScraper):
+    def __init__(self, url, proxies=None, timeout=None, *args, **kwargs):
+        super().__init__(url=url, proxies=proxies, timeout=timeout, *args, **kwargs)
+        scripts = self.soup.find_all("script")
+        recipe_id = None
+        for script in scripts:
+            matches = SCRIPT_PATTERN.search(str(script.string))
+            if matches:
+                data = matches.groupdict()
+                recipe_id = data["recipeId"]
+        data_url = f"https://proxy.api.coop.se/external/recipe/recipes/{recipe_id}?api-version=v1"
+        self.page_data = requests.get(
+            data_url,
+            proxies=proxies,
+            timeout=timeout,
+        ).content
+        self.data = json.loads(self.page_data)
+
+    @classmethod
+    def host(cls):
+        return "coop.se"
+
+    def author(self):
+        return self.data["recipeCreator"]
+
+    def title(self):
+        return self.data["name"]
+
+    def category(self):
+        for c in self.data["categories"]:
+            if len(c["categories"]) > 0:
+                if c["categories"][0] == "Typ av rätt":
+                    return c["categories"][1]
+        return None
+
+    def total_time(self):
+        return int(self.data["totalTime"].replace(" minuter", ""))
+
+    def yields(self):
+        return self.data["numberOfServings"]
+
+    def image(self):
+        return self.data["imageUrl"]
+
+    def ingredients(self):
+        groups = self.ingredient_groups()
+        ingredients = []
+        for g in groups:
+            ingredients.extend(g.ingredients)
+        return ingredients
+
+    def ingredient_groups(self):
+        groups = []
+        for p in self.data["recipePart"]:
+            ingredients = []
+            for i in p["ingredients"]:
+                quantity = i["quantity"]
+                if quantity.endswith(".0"):
+                    quantity = quantity[:-2]
+                unit = i["unit"]
+                if not unit == "":
```

Comment:
Nitpick: I think this can be simplified to:

```suggestion
                if unit:
```

(in other words: check the value of `unit` -- which should be a string value -- to see if it is truthy (non-empty), and if so prepend a space to its' contents)

Reply:
Much better